modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3960,7 +3961,10 @@ def groupby_agg(
                 by, agg_func, axis, groupby_kwargs, agg_args, agg_kwargs, how, drop
             )
 
-        if ExperimentalGroupbyImpl.get():
+        # 'group_wise' means 'groupby.apply()'. We're certain that range-partitioning groupby
+        # always works better for '.apply()', so we're using it regardless of the 'ExperimentalGroupbyImpl'
+        # value
+        if how == "group_wise" or ExperimentalGroupbyImpl.get():
```

Comment:
Let's rename `ExperimentalGroupbyImpl` to `RangePartitioningGroupby` in a separate PR as we discussed offline.